resources/2024-02-output-resources.md

```diff
@@ -0,0 +1,210 @@
+          {
+            "id": "/planes/kubernetes/local/namespaces/allinoneenv-allinoneapp/providers/core/ServiceAccount/allinonecontainer",
+            "localId": "ServiceAccount"
+          },
+          {
+            "id": "/planes/kubernetes/local/namespaces/allinoneenv-allinoneapp/providers/rbac.authorization.k8s.io/Role/allinonecontainer",
+            "localId": "KubernetesRole"
+          },
+          {
+            "id": "/planes/kubernetes/local/namespaces/allinoneenv-allinoneapp/providers/rbac.authorization.k8s.io/RoleBinding/allinonecontainer",
+            "localId": "KubernetesRoleBinding"
+          },
+          {
+            "id": "/planes/kubernetes/local/namespaces/allinoneenv-allinoneapp/providers/apps/Deployment/allinonecontainer",
+            "localId": "Deployment"
+          },
+          {
+            "id": "/planes/kubernetes/local/namespaces/allinoneenv-allinoneapp/providers/core/Service/allinonecontainer",
+            "localId": "Service"
+          }
+        ]
+      }
+      :
+]
+```
+
+### Additional enhancement
+
+We could also encapsulate the error message upon deployment of an output resource as one of its property. We could also map the provisioning status to the exact value to make it more informative. These two fields could then be used to to provide clearer communication to the user on what has gone wrong. 
+
+### Non goals
+
+* No changes to any of the existing APIs. The APIs will automatically pick up the changes.
+
+### User scenarios (optional)
+
+#### User story 1
+
+As a Radius User/Operator, I would like to get the information of a resource. If this resource provisioning has failed, I would like to know which of the out resources associated with the resource could not be deployed. 
+
+#### User story 2
+
+As a Radius User/Operator, I would like to be able to issue a `rad app delete` -a myapp to delete all resources provisioned and deployed by radius, even if the deployment of some of these resources had issues. 
+
+## Design
+
+Changes involve adding an attribute in resources.tsp, regenerating swagger files, handling the new attribute in convertors. 
+
+
+### API design
+
+***Model changes***
+
+Addition of provisioningStatus to OutputResource type in resources.tsp
+
+`typespec``
+
+@doc("Properties of an output resource.")
+model OutputResource {
+  @doc("The logical identifier scoped to the owning Radius resource. This is only needed or used when a resource has a dependency relationship. LocalIDs do not have any particular format or meaning beyond being compared to determine dependency relationships.")
+  localId?: string;
+
+  @doc("The UCP resource ID of the underlying resource.")
+  id?: string;
+
+  @doc("Determines whether Radius manages the lifecycle of the underlying resource.")
+  radiusManaged?: boolean;
+
+  @doc("Determines whether the resource was provisioned successfully")
+  provisioningStatus: ProvisioningStatus
+}
+
+@doc("Provisioning Status of Output Resource")
```

Comment:
Are these the only possible values? What will happen in the case where 2 output resources are rendered and the deployment of the 1st output resource fails? What will be the provisioningStatus of resource 2?

Reply:
Based on @rynowak 's feedback, I will just use the existing provisioning state enum with all its values. 
If only 1st outputResource failed, only its provisioning status will be failed. And the parent radius resource will have a failed provisioningStatus too. resource 2 will have its status based on its own deployment being successful.